File: app/src/main/java/me/toptas/rssreader/base/BaseMvpPresenter.java
package me.toptas.rssreader.base;

/**
 * Each presenter must implement this interface
 *
 * @param <V> View for the presenter
 */
public interface BaseMvpPresenter<V extends BaseView> {

    /**
     * Called when view attached to presenter
     *
     * @param view
     */
    void attach(V view);

    /**
     * Called when view is detached from presenter
     */
    void detach();

    /**
     * @return true if view is attached to presenter
     */
    boolean isAttached();
}

File: app/src/main/java/me/toptas/rssreader/base/BaseMvpPresenter.kt
package me.toptas.rssreader.base

/**
 * Each presenter must implement this interface
 *
 * @param <V> View for the presenter
</V> */
interface BaseMvpPresenter<V : BaseView> {

    /**
     * @return true if view is attached to presenter
     */
    val isAttached: Boolean

    /**
     * Called when view attached to presenter
     *
     * @param view
     */
    fun attach(view: V)

    /**
     * Called when view is detached from presenter
     */
    fun detach()
}